streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamPrint.java

```diff
@@ -62,13 +62,19 @@ public void init(ProcessorContext context) {
             if (valueSerde == null) {
                 this.valueSerde = context.valueSerde();
             }
-        }
 
-        @Override
-        public void process(final K key, final V value) {
-            final K deKey = (K) maybeDeserialize(key, keySerde.deserializer());
-            final V deValue = (V) maybeDeserialize(value, valueSerde.deserializer());
-            action.apply(deKey, deValue);
+            // no mapper provided, using a default one which takes care of checking byte[] for deserialization
+            if (mapper == null) {
+                mapper = new KeyValueMapper<K, V, String>() {
+                    @Override
+                    public String apply(K key, V value) {
+                        final K deKey = (K) maybeDeserialize(key, keySerde.deserializer());
```

Comment:
Here is my reasoning:

1. Ideally for any types of `K, V` we should be safely use their `toString()` overridden and the serdes would never be needed.

2. The ONLY exception, is that if `K, V` itself is `byte[]` (in practice key type `K` should never be `byte[]` either but be `Bytes` as it is needed for hashcode / compareTo). 

Note that for `KStreamPrint<K, V>` the passed in key and value should always be in `? extends K` and `? extends V`. In this case if the passed in key/value of the `process` call are really `byte[]` it means the `K` and `V` must be `byte[]`, then `Serde<byte[]>` is expected to be used which is a no-op anyways: maybe you can do a quick validation, I suspect that `KStream<byte[], byte[]>#print` today does not work at all.